gnome-shell/src/dark/meson.build

```diff
@@ -0,0 +1,83 @@
+# destination directory
+install_dir = join_paths(gnomeshell_alt_theme_dir, meson.project_name() + '-dark', 'gnome-shell')
+
+# generate .css files
+theme_sources = files([
+  'gnome-shell.scss',
+  '../gnome-shell-high-contrast.scss',
+  '../gnome-shell-sass/_colors.scss',
+  '../gnome-shell-sass/_common.scss',
+  '../gnome-shell-sass/_dock.scss',
+  '../gnome-shell-sass/_drawing.scss',
+  '../gnome-shell-sass/_high-contrast-colors.scss',
+  '../gnome-shell-sass/_ubuntu-colors.scss',
+])
+
+styles = [
+  'gnome-shell-high-contrast',
+  'gnome-shell'
+]
+
+theme_deps = []
+
+sassc = find_program('sassc')
+foreach style: styles
+  theme_deps += custom_target('style-' + style,
+                              input: '@0@.scss'.format(style),
+                              output: '@0@.css'.format(style),
+                              command: [
+                                sassc, '-a', '@INPUT@', '@OUTPUT@'
+                              ],
+                              install: true,
+                              install_dir: install_dir,
+                              depend_files: theme_sources)
+endforeach
+
+# static files to copy
+data_sources = [
+  '../calendar-today.svg',
+  '../checkbox-focused.svg',
+  '../checkbox-off-focused.svg',
+  '../checkbox-off.svg',
+  '../checkbox.svg',
+  '../checkbox-focused-dark.svg',
+  '../checkbox-off-focused-dark.svg',
+  '../checkbox-off-dark.svg',
+  '../checkbox-dark.svg',
+  '../dash-placeholder.svg',
+  '../key-enter.svg',
+  '../key-hide.svg',
+  '../key-layout.svg',
+  '../key-shift-latched-uppercase.svg',
+  '../key-shift.svg',
+  '../key-shift-uppercase.svg',
+  '../lockscreen-gradient.svg',
+  '../logged-in-indicator.svg',
+  '../meson.build',
+  '../message-indicator-symbolic.svg',
+  '../no-events.svg',
+  '../noise-texture.png',
+  '../no-notifications.svg',
+  '../pad-osd.css',
+  '../pointer-double-click-symbolic.svg',
+  '../pointer-drag-symbolic.svg',
+  '../pointer-primary-click-symbolic.svg',
+  '../pointer-secondary-click-symbolic.svg',
+  '../process-working.svg',
+  '../running-indicator.svg',
+  '../source-button-border.svg',
+  '../summary-counter.svg',
+  '../toggle-off-dark.svg',
+  '../toggle-off-hc.svg',
+  '../toggle-off-intl.svg',
+  '../toggle-off.svg',
+  '../toggle-on-dark.svg',
+  '../toggle-on-hc.svg',
+  '../toggle-on-intl.svg',
+  '../toggle-on.svg',
+  '../ws-switch-arrow-down.svg',
+  '../ws-switch-arrow-up.svg',
+]
+
+# install static data files
+install_data(data_sources, install_dir: install_dir)
```

Comment:
There is obviously a lot of duplication in this file.

My first impulse would usually be to factorize everything that is not the install_dir parameter. For example, to symlink the base build file and provide install_dir as a parameter or as part of an environment (I'm not familiarized with meson).

But in this case I believe this file could be seen as a place for little tweaks and hacks that could be required in the future to adjust the theme variant (for example, different icons), so in the end I have no strong opinion.

Reply:
it is easier if the install dir is shared

Reply:
I think I'll keep them separated tough. The `custom-target` directive now generates two different targets, one for light and one for dark, with the same name `gnome-shell.scss`, while keeping it in the same meson build is causing target duplication which resolution makes this even more complicated